docs/chains/example-transcribe.mdx

```diff
@@ -0,0 +1,275 @@
+
+```bash
+curl -X POST 'https://model<MODEL_ID>.api.baseten.co/development/predict' \
+    -H "Authorization: Api-Key $BASETEN_API_KEY" \
+    -d '<JSON_INPUT>'
+```
+
+with JSON input:
+
+```json
+{
+  "job_descr": {
+    "media_url": "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/TearsOfSteel.mp4",
+    "job_uuid": "3bcbfb53-0ffa-445e-824d-15012f5221db"
+  },
+  "params": {
+    "micro_chunk_size_sec": 30
+  }
+}
+```
+
+The output looks like this (truncated):
+
+```json
+{
+    "job_descr": {
+        "media_url": "http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/TearsOfSteel.mp4",
+        "job_uuid": "job_uuid_0",
+        "media_id": 0,
+        "status": "SUCCEEDED"
+    },
+    "segments": [
+        ...
+        {
+            "start_time_sec": 517.9465,
+            "end_time_sec": 547.70975,
+            "text": "The world's changed, Celia. Maybe we can too. Memory override complete!",
+            "language": "english",
+            "bcp47_key": "en"
+        },
+        {
+            "start_time_sec": 547.70975,
+            "end_time_sec": 567.0716874999999,
+            "text": "You know, there's a lesson to be learned from this. Could've gone worse.",
+            "language": "english",
+            "bcp47_key": "en"
+        },
+        ...
+    ],
+    "input_duration_sec": 734.261406,
+    "processing_duration_sec": 82.42135119438171,
+    "speedup": 8.908631020478238
+}
+```
+
+The `Transcribe` chainlet does the following:
+* Assert that the media URL supports range downloads. This is usually a given
+  for video / audio hosting services.
+* Uses `FFMPEG` to query the length of the medium (both video and audio
+  files are supported).
+* Generates a list of "macro chunks", defined by their start and end times.
+  The length is defined by `macro_chunk_size_sec` in `TranscribeParams`.
+  TODO: this will soon be upgraded to find silence aware split points, so
+  that a chunk does not end in the middle of a spoken word. To do this a
+  small segment around the desired chunk boundary is downloaded (e.g. +/- 5
+  seconds) and the most silent timestamp within is determined.
+* Sends the media URL with chunk limits as "tasks" to `MacroChunkWorker`. Using
+  `asyncio.ensure_future`, these tasks are dispatched concurrently - meaning
+  that the loop over the chunks does not wait for each chunk to complete first,
+  before dispatched the task on the next chunk. These "calls" are network
+  requests (RPCs) to the `MacroChunkWorker` chainlet which runs on its own
+  deployment and can auto-scale, depending on the load.
+* Once all tasks are dispatched, it waits for the results and concatenates
+  all the partial transcriptions from the chunks to a final output.
+
+### `MacroChunkWorker`
+
+The `MacroChunkWorker` chainlet works on chunk tasks it receives from the
+`Transcribe` chainlet. For each chunk it does the following:
+
+* It starts a `DownloadSubprocess` asynchronously (i.e. this will need CPU on
```

Comment:
I think breaking this up some code samples could be good -- this is quite a bit of text. It also might be a bit more detail than the reader needs